internal/pkg/manifest/pipeline.go

```diff
@@ -241,16 +263,12 @@ func UnmarshalPipeline(in []byte) (*Pipeline, error) {
 	return nil, errors.New("unexpected error occurs while unmarshalling manifest.yml")
 }
 
-// IsCodeStarConnection indicates to the manifest if this source requires a CSC connection.
-func (s Source) IsCodeStarConnection() bool {
-	switch s.ProviderName {
-	case GithubProviderName:
+// NeedsCodeStarConnection indicates to the manifest if this source requires a CSC connection.
+func (s Source) NeedsCodeStarConnection() bool {
+	if (s.ProviderName == GithubProviderName || s.ProviderName == BitbucketProviderName) && s.Connection == "" {
```

Comment:
dang this is tricky, I'd love not to have to repeat logic - we're kind of duplicating when to ask for codestar in both `pipeline_init` and here 🤔